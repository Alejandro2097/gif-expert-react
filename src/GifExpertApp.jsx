import { useState } from "react"

export const GifExpertApp = () => {
  const [categories, setCategories] = useState(['one punch', 'Dragon ball']);
  const onAddCatecory = () => {
    setCategories([...categories, 'Resident evil 4'])
  }

  console.log(categories);
  return (
    <>
        {/* Titulo */}
        <h1>GifExpertApp</h1>
        {/* Input */}
        <input/>
        {/* Listado de gifs */}
        <button onClick={onAddCatecory}>Agregar</button>
        <ol>
          {categories.map(category => {
              return <li key={category}>{category}</li>
              })
           }
        </ol>
    </>
  )
}
