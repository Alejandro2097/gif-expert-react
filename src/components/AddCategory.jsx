import PropTypes from 'prop-types';
import { useState } from "react";

export const AddCategory = ({onNewCategory}) => {
    const [inputValue, setInputValue] = useState();
    const onInputChange = ({target}) => {
        setInputValue(target.value)
    }
    const onSubmit = (event) => {
        console.log('Hola mundo desde onSubmit');
        event.preventDefault();
        if(inputValue.trim().length <= 1) return;
        // setCategories(categories => [inputValue, ...categories]);
        onNewCategory(inputValue.trim());
        setInputValue('');
    }
  return (
    <form onSubmit={(event) => onSubmit(event)} aria-label="form">
        <input
        type="text"
        placeholder="Buscar Gifs"
        value={inputValue}
        onChange={(event) => onInputChange(event)}
    />
    </form>
    
  )
}


AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired
}