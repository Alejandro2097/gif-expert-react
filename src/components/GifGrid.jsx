export const GifGrid = ({category}) => {

  const getGifs = async() => {
    const url = `https://api.giphy.com/v1/gifs/search?api_key=qg1pvYB3j3MRWnaXQfR3KHi67UuXJTch&q=${category}`;
    const resp = await fetch(url);
    console.log(resp);
  }
  return (
    <>
        <h3>{category}</h3>
        <p>Hola mundo</p>
    </>
  )
}
