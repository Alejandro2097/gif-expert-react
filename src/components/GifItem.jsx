export const GifItem = (props) => {
  console.log(props);
  return (
    <div>ahh</div>
  )
}
