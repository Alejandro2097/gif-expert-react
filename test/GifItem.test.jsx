import { render } from "@testing-library/react";

import { GifItem } from "../src/components";

describe('Pruebas en <GifItem/>',  () => {
    const url = 'https://media3.giphy.com/media/ITol7gnfFqLEHGpG0t/giphy-downsized-medium.gif?cid';
    const title = 'Resident evil'
    test('debe verificar las propTypes', () => {
        const { container } = render(<GifItem url={url} title={title}/>)
        expect(container).toMatchSnapshot();
    });

    test('debe de mostrar la imagen y la URL y ALT indicada', () => {
        render(<GifItem title={title} url={url}/>);
        // screen.debug();
        // expect(screen.getByRole('img').src).toBe(url);
        // expect(screen.getByRole('img').src).toBe(title);
        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url)
        expect(src).toBe(alt)
    });

    test('debe de mostrar el titulo en el componente', () => {
        render(<GetItem title={title} url={url}/>);
        expect(screen.getByText(title)).toBeTruthy();
    })
});