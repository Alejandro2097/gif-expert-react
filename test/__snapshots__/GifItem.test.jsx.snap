// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem/> debe verificar las propTypes 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Resident evil"
      src="https://media3.giphy.com/media/ITol7gnfFqLEHGpG0t/giphy-downsized-medium.gif?cid"
    />
    <p>
      Resident evil
    </p>
  </div>
</div>
`;
